pkg/runctl/run_manager.go

```diff
@@ -133,13 +133,15 @@ func (c *runManager) prepareRunNamespace(ctx *runContext) error {
 		return errors.Wrap(err, "failed to create main run namespace")
 	}
 	ctx.pipelineRun.UpdateRunNamespace(ctx.runNamespace)
+	ctx.pipelineRun.CommitStatus()
```

Comment:
Why returned error is swallowed?

Reply:
Now not swallowed anymore

Reply:
But now there are too many error cases to handle at once. Why not just fail fast?

Reply:
We apply the changes directly when calling the update methods to the in-memory representation of the pipeline run. In case that results in an error we will get the error at that point in time. From that point of view we are fail fast. In case the later push fails we retrieve the updated version of the pipeline run and re-apply the changes. In case we fail now we get errors from applying the changes or from performing the update ... But how should we be failfast in that case?